### PR TITLE
XM-VESG-1 Flamer Nozzle or: How I Learned to Stop Worrying and Love the Burning Naphtha

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Adds the XM-VESG-1 Flamer Nozzle, an attachment for the standard M240A1 incinerator unit. It allows the flamethrower to launch a fireball up to five tiles away from the user. It burns three units of fuel each use, and has a three second cooldown, independent of the flamethrower's.

Resprites the M240A1 and M240-T incinerator units, by Triiodine.

Special thanks to Kaga and Sargeantmuffinman for helping me get the name and description sorted.

## Why It's Good For The Game

It's pretty cool and gives mini-pyros the ability to blast something without leaving a trail of fire in the way.

## Changelog

:cl: Geeves, Triiodine
add: Adds the XM-VESG-1 Flamer Nozzle, an attachment for the standard M240A1 incinerator unit. Available in Req and Squad-Req as an underbarrel attachment.
imageadd: Resprites the M240A1 and M240-T incinerator units.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun_attachables.dm [hunk 1]
```diff
@@ -1924,9 +1940,12 @@ Defined in conflicts.dm of the #defines folder.
...
 	..()
 	if(internal_extinguisher)
-		to_chat(user, "It contains [internal_extinguisher.reagents.total_volume] units of water left!")
+		to_chat(user, SPAN_NOTICE("It has [internal_extinguisher.reagents.total_volume] unit\s of water left!"))
 		return
-	to_chat(user, "It's empty.")
+	to_chat(user, SPAN_WARNING("It's empty."))
+
+/obj/item/attachable/attached_gun/extinguisher/handle_attachment_description(var/slot)
+	return "It has [icon2html(src)] [name] ([current_rounds]/[max_rounds]) mounted underneath.<br>"
```
Comment (Segrain) [inline code/modules/projectiles/gun_attachables.dm:1948]:
```suggestion
	return "It has [icon2html(src)] [name] ([internal_extinguisher.reagents.total_volume]/[internal_extinguisher.max_water]) mounted underneath.<br>"
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]
fix being able to chungus the flamer

File: code/modules/projectiles/gun_attachables.dm [hunk 2]
```diff
@@ -1936,9 +1952,12 @@ Defined in conflicts.dm of the #defines folder.
...
 	..()
 	if(internal_extinguisher)
-		to_chat(user, "It contains [internal_extinguisher.reagents.total_volume] units of water left!")
+		to_chat(user, SPAN_NOTICE("It has [internal_extinguisher.reagents.total_volume] unit\s of water left!"))
 		return
-	to_chat(user, "It's empty.")
+	to_chat(user, SPAN_WARNING("It's empty."))
+
+/obj/item/attachable/attached_gun/extinguisher/handle_attachment_description(var/slot)
+	return "It has [icon2html(src)] [name] ([internal_extinguisher.reagents.total_volume]/[internal_extinguisher.max_water]) mounted underneath.<br>"
```
Comment (stanalbatross) [inline code/modules/projectiles/gun_attachables.dm:1960]:
```suggestion
	return "It has \a [icon2html(src)] [name] ([internal_extinguisher.reagents.total_volume]/[internal_extinguisher.max_water]) mounted underneath.<br>"
```

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -1967,6 +1986,82 @@ Defined in conflicts.dm of the #defines folder.
...
+	var/fuel_per_projectile = 3
+
+	var/static/list/fire_sounds = list(
+		'sound/weapons/gun_flamethrower1.ogg',
+		'sound/weapons/gun_flamethrower2.ogg',
+		'sound/weapons/gun_flamethrower3.ogg'
+	)
+
+/obj/item/attachable/attached_gun/flamer_nozzle/handle_attachment_description(var/slot)
+	return "It has [icon2html(src)] [name] mounted beneath the barrel.<br>"
```
Comment (stanalbatross) [inline code/modules/projectiles/gun_attachables.dm:2015]:
```suggestion
	return "It has \a [icon2html(src)] [name] mounted beneath the barrel.<br>"
```

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -1967,6 +1986,82 @@ Defined in conflicts.dm of the #defines folder.
...
+	ammo_datum.flamer_reagent_type = flamer_reagent.type
+	P.generate_bullet(ammo_datum)
+	P.icon_state = "naptha_ball"
+	P.color = flamer_reagent.color
+	P.fire_at(target, user, user, max_range, AMMO_SPEED_TIER_2, null, FALSE)
+	var/turf/user_turf = get_turf(user)
+	gun.muzzle_flash(Get_Angle(user_turf, target), user)
+	playsound(user_turf, pick(fire_sounds), 50, TRUE)
+
+	to_chat(user, SPAN_WARNING("The gauge reads: <b>[round(gun.current_mag.get_ammo_percent())]</b>% fuel remains!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/gun_attachables.dm:2063]:
```suggestion
	to_chat(user, SPAN_WARNING("The gauge reads: <b>[round(gun.current_mag.get_ammo_percent())]</b>% fuel remaining!"))
```

File: code/modules/projectiles/gun_attachables.dm [hunk 5]
```diff
@@ -1967,6 +1986,82 @@ Defined in conflicts.dm of the #defines folder.
...
+
+	var/obj/item/projectile/P = new(src, create_cause_data(initial(name), user, src))
+	var/datum/ammo/flamethrower/ammo_datum = new projectile_type
+	ammo_datum.flamer_reagent_type = flamer_reagent.type
+	P.generate_bullet(ammo_datum)
+	P.icon_state = "naptha_ball"
+	P.color = flamer_reagent.color
+	P.fire_at(target, user, user, max_range, AMMO_SPEED_TIER_2, null, FALSE)
+	var/turf/user_turf = get_turf(user)
+	gun.muzzle_flash(Get_Angle(user_turf, target), user)
```
Comment (stanalbatross) [inline code/modules/projectiles/gun_attachables.dm:2060]: does the flamer get one?

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by Sargeantmuffinman** [comment]
The range is of it is too low.
4 tiles isn't enough to have the flamethrower attachment turn it into an offensive weapon.
Having it being able to be scaled by the fuel itself would be cool but probably out of scope of this PR.

I suggest upping the range to 6 tiles as that's one less than the 7 tile screen range.

**Comment by Sargeantmuffinman** [comment]
It doesn't work with custom fuel.

**Comment by Geevies** [comment]
> It doesn't work with custom fuel.

as far as i can tell, all the reagent code checks out, so it's something to do with custom chems, which is out of scope for this PR

File: code/modules/projectiles/gun_attachables.dm [hunk 6]
```diff
@@ -2032,6 +2051,82 @@ Defined in conflicts.dm of the #defines folder.
...
```
Comment (stanalbatross) [inline code/modules/projectiles/gun_attachables.dm:2125]: this just makes it do a generic muzzle flash when firing so either make a unique sprite (greyscale and colourswap based on reagent colour) or don't use it

File: code/modules/projectiles/guns/flamer/flamer.dm [hunk 7]
```diff
@@ -217,6 +220,7 @@
 	if(fire)
 		qdel(fire)
 
+	muzzle_flash(Get_Angle(get_turf(user), target), user)
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/flamer/flamer.dm:223]: same

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
> It doesn't work with custom fuel.

worked fine with a vved hydrogen on local

**Comment by Sargeantmuffinman** [comment]
> > It doesn't work with custom fuel.
> 
> worked fine with a vved hydrogen on local

Try using a research generated one. Probably should have specified that we used that.

**Comment by Ticker175** [comment]
Sorry for the late comment but I was testing the nossle a week or so ago with coustom research chems and discovered that with the nossle attachment enabled, coustom chems do not work at all.

With a coustom chem tank loaded and the nossle enabled, the flamer will fire a projectile that creates a 1x1 normal fire where it lands for about half a second then it goes out. Tested with multiple diffrent research chems over multiple rounds


another thing that is most likely a research issue but I’ll mention nonetheless is that coustom chems are VERY weak and desperately need a buff, not only is their damage terrible but the effects of flowing is so so so limited with a max range of 4, the effects of viscious make the flamer 1 tile and there for literally unusable (it does (apparently) increase damage but it’s not even noticeable), the effects of fueling and oxidising are also extremely weak and fire pen is so weak it’s hard to even tell if it does ANY damage to fire proof xenos

**Review by stanalbatross — APPROVED** [review]

**Review by nauticall — APPROVED** [review]
Sprites look a-ok to me 👍

**Comment by Nanu308** [comment]
Might see future tweaks to range or its availability, approved.

**Pull request merged**